include/libtorrent/alert_types.hpp

```diff
@@ -2663,11 +2663,27 @@ namespace libtorrent {
 		aux::allocation_slot m_v6_nodes_idx;
 	};
 
+	struct TORRENT_EXPORT block_uploaded_alert final : peer_alert
+	{
+		// internal
+		block_uploaded_alert(aux::stack_allocator& alloc, torrent_handle h
+			, tcp::endpoint const& ep, peer_id const& peer_id, int block_num
+			, int piece_num);
+
+		TORRENT_DEFINE_ALERT(block_uploaded_alert, 94)
+
+		static const int static_category = alert::progress_notification;
+		virtual std::string message() const override;
+
+		int block_index;
+		int piece_index;
```

Comment:
when deprecated functions are disabled, the type-safety is enforced, and ``piece_index_t`` does not implicitly convert to ``int`` anymore. https://ci.appveyor.com/project/arvidn/libtorrent/build/6561/job/9vcf5vjw7ubjmntq#L204